src/sage/libs/ntl/ntl_ZZ_pX.pyx

```diff
@@ -501,16 +502,67 @@ cdef class ntl_ZZ_pX():
             sage: g = ntl.ZZ_pX([-1,0,1],c)
             sage: g**10
             [1 0 10 0 5 0 0 0 10 0 8 0 10 0 0 0 5 0 10 0 1]
+
+            sage: x = ntl.ZZ_pX([0,1],c)
+            sage: x**10
+            [0 0 0 0 0 0 0 0 0 1]
         """
+        if modulus is None:
+            return self._pow(n)
+        else:
+            return self._powmod(n, modulus)
+
+    def _pow(ntl_ZZ_pX self, n):
         if n < 0:
             raise NotImplementedError
+        cdef long ln = n
         #self.c.restore_c() # restored in _new()
         cdef ntl_ZZ_pX r = self._new()
+        if self.is_x() and n >= 1:
+            ZZ_pX_LeftShift(r.x, self.x, n-1)
+        else:
+            sig_on()
+            ZZ_pX_power(r.x, self.x, ln)
+            sig_off()
+        return r
+
+    def _powmod(ntl_ZZ_pX self, n, ntl_ZZ_pX modulus):
+        """
+        Compute the n-th power of self modulo a polynomial.
```

Comment:
```suggestion
    cdef _powmod(ntl_ZZ_pX self, n, ntl_ZZ_pX modulus):
        """
        Compute the n-th power of self modulo a polynomial.
```
although with making it a `cdef`, you would remove the doctests explicitly calling the method.